dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/ingest/IngestUtilTest.java

```diff
@@ -6,320 +6,173 @@
-        datafile2.setChecksumType(DataFile.ChecksumType.SHA1);
-        datafile2.setChecksumValue("Unknown");
-
-        // set metadata and add version
-        FileMetadata fmd2 = new FileMetadata();
-        fmd2.setId(2L);
-        fmd2.setLabel("datafile2.txt");
-        fmd2.setDataFile(datafile2);
+        DataFile datafile2 = createDataFile(dataset, "application/octet-stream", "datafile2.txt");
+        FileMetadata fmd2 = createFileMetadata(datasetVersion, datafile2, 2L, "datafile2.txt");
         datafile2.getFileMetadatas().add(fmd2);
         datasetVersion.getFileMetadatas().add(fmd2);
-        fmd2.setDatasetVersion(datasetVersion);
 
         dataFileList.add(datafile2);
 
+        //WHEN
         IngestUtil.checkForDuplicateFileNamesFinal(datasetVersion, dataFileList);
 
-        boolean file1NameAltered = false;
-        boolean file2NameAltered = false;
-        for (DataFile df : dataFileList) {
-            if (df.getFileMetadata().getLabel().equals("datafile1-1.txt")) {
-                file1NameAltered = true;
-            }
-            if (df.getFileMetadata().getLabel().equals("datafile2-1.txt")) {
-                file2NameAltered = true;
-            }
-        }
-
-        // check filenames are unique and unaltered
-        assertEquals(file1NameAltered, true);
-        assertEquals(file2NameAltered, true);
+        //THEN
+        assertAll(
+                () -> assertTrue(dataFileList.stream().anyMatch(fileMetadataHasLabel("datafile1-1.txt"))),
+                () -> assertTrue(dataFileList.stream().anyMatch(fileMetadataHasLabel("datafile2-1.txt")))
+        );
 
         // try to add data files with "-1" duplicates and see if it gets incremented to "-2"
         IngestUtil.checkForDuplicateFileNamesFinal(datasetVersion, dataFileList);
 
-        for (DataFile df : dataFileList) {
-            if (df.getFileMetadata().getLabel().equals("datafile1-2.txt")) {
-                file1NameAltered = true;
-            }
-            if (df.getFileMetadata().getLabel().equals("datafile2-2.txt")) {
-                file2NameAltered = true;
-            }
-        }
-
-        // check filenames are unique and unaltered
-        assertEquals(file1NameAltered, true);
-        assertEquals(file2NameAltered, true);
+        assertAll(
+                () -> assertTrue(dataFileList.stream().anyMatch(fileMetadataHasLabel("datafile1-2.txt"))),
+                () -> assertTrue(dataFileList.stream().anyMatch(fileMetadataHasLabel("datafile2-2.txt")))
+        );
     }
 
-    @Test
-    /**
-     * Test adding duplicate file name labels to a dataset version with empty
-     * directory labels. This should simulate what happens when uploading a file
-     * via the file upload UI.
-     */
-    public void testCheckForDuplicateFileNamesWithEmptyDirectoryLabels() throws Exception {
+    @NotNull
+    private FileMetadata createFileMetadata(DatasetVersion datasetVersion, DataFile datafile1, long l, String s) {
```

Comment:
I forgot to tell you, we are generally putting private methods at the bottom, so you can read tests 1 by 1.

Reply:
Moved to bottom